src/Azure/Azure.Quantum.Client.Test/WorkspaceTest.cs

```diff
@@ -3,246 +3,291 @@
-                QuantumClient = new MockHelper.MockQuantumClient(httpMock)
+                Id = jobId,
+                Name = "Azure.Quantum.Unittest",
+                OutputDataFormat = "microsoft.qio-results.v2",
+                InputParams = new Dictionary<string, object>()
                 {
-                    SubscriptionId = TestConstants.SubscriptionId,
-                    ResourceGroupName = TestConstants.ResourceGroupName,
-                    WorkspaceName = TestConstants.WorkspaceName,
-                    BaseUri = new Uri(TestConstants.Endpoint),
+                    { "params", new Dictionary<string, object>() },
                 },
+                InputDataUri = inputUri,
             };
         }
 
-        private void SetJobResponseMessage(string jobId)
+        private static Problem CreateTestProblem()
         {
-            this.httpMock.ResponseMessage = new HttpResponseMessage(HttpStatusCode.OK)
+            // Create an Ising-type problem for shipping-containers
+            var containerWeights = new int[] { 1, 5, 9, 21, 35, 5, 3, 5, 10, 11 };
+
+            var problem = new Problem(ProblemType.Ising);
+            for (int i = 0; i < containerWeights.Length; i++)
             {
-                Content = new StringContent(JsonConvert.SerializeObject(CreateJobDetails(jobId))),
-            };
-        }
+                for (int j = 0; j < containerWeights.Length; j++)
+                {
+                    if (i != j)
+                    {
+                        problem.Add(i, j, containerWeights[i] * containerWeights[j]);
+                    }
+                }
+            }
 
-        private static JobDetails CreateJobDetails(string jobId)
-        {
-            return new JobDetails(
-                id: jobId,
-                containerUri: "https://uri",
-                inputDataFormat: "format1",
-                providerId: TestConstants.ProviderId,
-                target: "target");
+            return problem;
         }
 
-        private void ValidateJobRequestMessage(
-            string jobId,
-            params HttpMethod[] methods)
+        private async Task<(string, string)> UploadProblem(IWorkspace workspace, Problem problem, string jobId)
         {
-            var requestMessages = httpMock.RequestMessages;
-            Assert.AreEqual(requestMessages.Count, methods.Length);
+            string intermediaryFile = Path.GetTempFileName();
+
+            // Save to the intermediary file
+            using (var intermediaryWriter = File.Create(intermediaryFile))
+            {
+                using (var compressionStream = new GZipStream(intermediaryWriter, CompressionLevel.Fastest))
+                {
+                    await problem.SerializeAsync(compressionStream);
+                }
+            }
 
-            for (var i = 0; i < requestMessages.Count; i++)
+            using (var intermediaryReader = File.OpenRead(intermediaryFile))
```

Comment:
Nit, but may be able to simplify with the new `using var` notation since the `using` scope ends at the same point as the enclosing scope.